### PR TITLE
Visualizing the KS test results

### PR DESCRIPTION
**Pull request by gwaybio**
Here is a notebook that visualizes the KS test analysis, plus one example distribution.

In addition to carefully reviewing the PR, please also let me know the command you have been using to `nbconvert` the `ipynb` file so that we can keep internally consistent.

### CODE REVIEW

**Comment by jenna-tomkinson** [comment]
@gwaybio 
The command I always use is:
```
jupyter nbconvert --to python notebook.ipynb
```

I can now see why it is easier to review a `.py` file.

**Comment by gwaybio** [comment]
done, @jenna-tomkinson back to you!

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+
+suppressPackageStartupMessages(library(ggplot2))
+suppressPackageStartupMessages(library(dplyr))
+
+
+# In[2]:
+
+
+output_fig_dir <- "figures"
```
Comment (jenna-tomkinson) [inline 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: Is this whole thing written in R language but in a python file?

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Visualize the results of the KS-test
+
+# In[1]:
+
+
+suppressPackageStartupMessages(library(ggplot2))
+suppressPackageStartupMessages(library(dplyr))
```
Comment (jenna-tomkinson) [inline 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: Is this importing the libraries?

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+ks_test_fig <- file.path(output_fig_dir, "ks_test_cp_genotype.png")
+actin_distrib_fig <- file.path(output_fig_dir, "top_actin_feature_distrib.png")
+
+
+# In[3]:
+
+
+# Load and process KS-test data
+ks_file <- file.path("data", "nf1_kstest_two_sample_results.csv")
+ks_df <- readr::read_csv(ks_file, col_types = readr::cols(.default="d", Features="c"))
```
Comment (jenna-tomkinson) [inline 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: `(readr::cols(.default="d", Features="c")))`

What do the letters mean in this case?

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+print(dim(cp_df))
+head(cp_df, 3)
+
+
+# ## Visualize KS test
+
+# In[5]:
+
+
+# Determine a bonferroni adjusted alpha value threshold
```
Comment (jenna-tomkinson) [inline 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: Why use this to determine the threshold? Is there a reason or is this the standard?

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+# ## Visualize distribution of top actin feature
+
+# In[8]:
+
+
+top_actin_feature <- ks_df %>%
+    dplyr::filter(channel_cleaned == "actin") %>%
+    dplyr::select(Features)
+
+top_actin_feature <- head(top_actin_feature, 1) %>% dplyr::pull(Features)
```
Comment (jenna-tomkinson) [inline 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: What does it mean to be a "top actin feature"? Is this the feature from the actin channel with the lowest p-value from the ks-test?

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! All my comments are questions in regard to the process used for this so I can get a better understanding for future use in other projects.

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 1 above)
Comment (gwaybio) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: good catch! it is supposed to be a `.r` file. The command `jupyter nbconvert --to python notebook.ipynb` is not compatible with r notebooks. I used this instead: `jupyter nbconvert --to=script --FilesWriter.build_directory=. *.ipynb`

Comment (gwaybio) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: (see e5fc843707bfa7004d5c5ab9f7f66e7516c0838e)

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 2 above)
Comment (gwaybio) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: yep! (the `library()`) syntax is equivalent to import statements in python)

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 3 above)
Comment (gwaybio) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: d refers to `double` and c refers to `character`. It is similar to the `dtype` options that pandas has for columns. R does a better job at handling data types more explicitly than python!

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 4 above)
Comment (gwaybio) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: that is correct! we need to adjust our alpha value for something called "multiple test correction". Bonferroni is the simplest and most straightfoward answer. Just divide 0.05 by the number of comparisons!

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 5 above)
Comment (gwaybio) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: yep - you can see exactly what I'm doing to sort the values above, which will tell you what the top feature would represent.

**Comment by gwaybio** [comment]
Thanks for the review @jenna-tomkinson - I am going to merge!

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 1 above)
Comment (jenna-tomkinson) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: Oh interesting! This is very good to know!

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 2 above)
Comment (jenna-tomkinson) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: Ahh okay so this is the import libraries for R! I am now getting this whole file was an R file.

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 3 above)
Comment (jenna-tomkinson) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: After dealing with the dtypes when trying to work with the `.transpose` function in python, I can see how convenient this is!

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 4 above)
Comment (jenna-tomkinson) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: That is very simple! Good to know there are multiple ways of doing this also but this simple version seems to be the best option for this project.

File: 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py (hunk 5 above)
Comment (jenna-tomkinson) [reply 5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py]: After discussion at yesterday's meeting, this makes total sense!

**Pull request merged**